### ISSUE TITLE
Unknown error getting FCM Token - Please set your project ID

### ISSUE THREAD

**Issue by NeohWei97** [OneSignal/OneSignal-Android-SDK#1017]
**Description:**
I have encountered an issue during OneSignal initialization, saying that there is an error getting the FCM token, and that I would need to set my Firebase Project ID in order to communicate with Firebase server APIS. However when I checked the google-services.json file, the project_id is already set.

**Environment**
1. I am using OneSignal Android SDK 3.12.4
2. The SDK is added through maven
3. My project is using other firebase services (Firebase Storage, Firebase Auth)

**Steps to Reproduce Issue:**
  1. Add the OneSignal SDK to your project through gradle
  2. Add following Firebase dependencies in build.gradle(app)
      - implementation 'com.google.firebase:firebase-core:17.4.1'
      - implementation 'com.google.firebase:firebase-storage:19.1.1'
      - implementation 'com.google.firebase:firebase-firestore:21.4.3'
      - implementation 'com.google.firebase:firebase-auth:19.3.1'
      - implementation 'com.google.firebase:firebase-messaging:20.1.7'
  3. Added google-services.json to project
  4. Initialize OneSignal in application class.
  5. On run, you would get an error message from OneSignal with the message "Unknown error getting FCM Token".

**Anything else:**

(crash stacktraces, as well as any other information here)
Error message:
E/OneSignal: Unknown error getting FCM Token
    java.lang.IllegalArgumentException: Please set your project ID. A valid Firebase project ID is required to communicate with Firebase server APIs: It identifies your project with Google.
        at com.google.android.gms.common.internal.Preconditions.checkNotEmpty(Unknown Source:8)
        at com.google.firebase.iid.FirebaseInstanceId.zza(com.google.firebase:firebase-iid@@20.1.7:52)
        at com.google.firebase.iid.FirebaseInstanceId.getInstance(com.google.firebase:firebase-iid@@20.1.7:2)
        at com.onesignal.PushRegistratorFCM.getToken(PushRegistratorFCM.java:84)
        at com.onesignal.PushRegistratorAbstractGoogle.attemptRegistration(PushRegistratorAbstractGoogle.java:97)
        at com.onesignal.PushRegistratorAbstractGoogle.access$100(PushRegistratorAbstractGoogle.java:37)
        at com.onesignal.PushRegistratorAbstractGoogle$1.run(PushRegistratorAbstractGoogle.java:84)
        at java.lang.Thread.run(Thread.java:919)

**Comment by jkasten2** [comment]
@NeohWei97 This is a known compatibility issue that was fixed in OneSignal Android SDK 3.12.7, however using 3.13.0+ is highly recommended.

I have created a pin issue https://github.com/OneSignal/OneSignal-Android-SDK/issues/1018 to surface this issue further so it is not overlooked.

Thanks for reporting, as it indicates we don't have enough documentation around this.

@NeohWei97 Was this an app you already released and are just updating Firebase? Or did you integrate OneSignal for the first time? We are wondering so we can improve our documentation around this.
